qgis_deployment_toolbelt/jobs/job_profiles_synchronizer.py

```diff
@@ -88,13 +92,79 @@ def run(self) -> None:
             f"{', '.join(self.PROFILES_NAMES_DOWNLOADED)}"
         )
 
+        # filter out profiles that do not match the rules
+        profiles_matched, profiles_unmatched = self.filter_profiles_on_rules(
+            li_downloaded_profiles=profiles_folders
+        )
+        if not len(profiles_matched):
+            logger.warning(
+                "None of the downloaded profiles meet the deployment requirements."
+            )
+            return
+
+        logger.info(
+            f"Of the {len(profiles_folders)} profiles downloaded, "
+            f"{len(profiles_unmatched)} do not meet the conditions for deployment."
+        )
+
         # copy profiles to the QGIS 3
         self.sync_installed_profiles_from_downloaded_profiles(
-            downloaded_profiles=profiles_folders
+            downloaded_profiles=profiles_matched
         )
 
         logger.debug(f"Job {self.ID} ran successfully.")
 
+    def filter_profiles_on_rules(
+        self, li_downloaded_profiles: Iterable[QdtProfile]
+    ) -> tuple[list[QdtProfile], list[QdtProfile]]:
+        """Evaluate profile regarding to its deployment rules.
+
+        Args:
+            li_downloaded_profiles (Iterable[QdtProfile]): input list of QDT profiles
+
+        Returns:
+            tuple[list[QdtProfile], list[QdtProfile]]: tuple of profiles that matched
+            and those which did not match their deployment rules
+        """
+        li_profiles_matched = []
+        li_profiles_unmatched = []
+
+        context_object = {"environment": environment_dict()}
+        for profile in li_downloaded_profiles:
+            if profile.rules is None:
+                logger.debug(f"No rules to apply to {profile.name}")
+                li_profiles_matched.append(profile)
+                continue
+
+            logger.debug(
+                f"Checking that profile '{profile.name}' matches deployment conditions."
+                f"{len(profile.rules)} rules found."
+            )
+            try:
+                engine = RuleEngine(rules=profile.rules)
+                results = engine.evaluate(obj=context_object)
```

Comment:
Does it mean that we can only use the values defined in the previously defined `context_object` ?

Let's say I wanted to create a rule:

```json
{
      "name": "Environnement variable",
      "description": "Profile is configured to run only if an environnement variable is defined.",
      "conditions": {
        "all": [
          {
            "path": "os.env.get('QDT_PROFILE_FOO_ENABLED','False')"
            "value": "True",
            "operator": "equal"
          }
        ]
      }
    }
```

Could it work ?

Reply:
We need a new PR to provide the environment variable. This will be done later.